packages/kit-headless/src/components/popover/popover-trigger.tsx

```diff
@@ -18,22 +18,24 @@ import { PopoverContext } from './popover-context';
 import styles from './popover-trigger.css?inline';
 
 export const PopoverTrigger = component$(
-  (props: ExtendedPropsByAriaAttribute) => {
+  (props: ExtendedPropsByAriaAttribute<'span'>) => {
     const ref = useSignal<HTMLElement>();
     const contextService = useContext(PopoverContext);
     useStylesScoped$(styles);
-    const store = useStore<Partial<QwikUiAreaAttributesFunctionReturnType>>({
+    const ariaAttributesStore = useStore<
+      Partial<QwikUiAreaAttributesFunctionReturnType<'span'>>
+    >({
       lastKey: undefined,
-      ariaAttributes: {},
+      ariaAttributes: undefined,
     });
     useTask$(({ track }) => {
       track(() => ({ ...props }));
-      const { lastKey, ariaAttributes } = getAriaAttributes(
+      const { lastKey, ariaAttributes } = getAriaAttributes<'span'>(
         props,
-        store.lastKey
+        ariaAttributesStore.lastKey
       );
-      store.ariaAttributes = ariaAttributes;
-      store.lastKey = lastKey;
+      ariaAttributesStore.ariaAttributes = ariaAttributes;
+      ariaAttributesStore.lastKey = lastKey;
```

Comment:
I'm not sure why do we need an extra store, we can spread the props over the element itself instead (like we do in the tabs or accordion) which will remove the unnecessary management code.

Reply:
I think we need because of snake- and camel-case of the keys?

Reply:
but in Qwik it's not an issue, people are not suppose to write their attributes with snake case or camel etc, only kebab, just like standard HTML

Reply:
Okay understand though didn't try that out. Maybe you can fix it yourself. I guess it would be better to limit to the aria attributes instead of using the Qwik type? The types to achieve that should be there. Or should I do it?

Reply:
Okay I just quickly tried it. It works. I don't personally like it, because this kebab-case what I wrongly named snake-case has lead to different problems over time when you want to use JavaScript-Objects and are forced for the bracket notation. But he that is just personal taste.

 I will try to add your solution tomorrow with picking the aria-labels out of the Intrinsic-Elements and leave them to kebab case. Should we keep the generic which extends the normal props or should we plainly go with a helper type to extract them?
 
 Will medidate now and then go to sleep.
 
 GN8